README.md

```diff
@@ -578,33 +599,56 @@ Note that the key is the parameter we defined in our route, and the value is
 what appears in the URL.
 
 We can now use the data contained in our params object to access the specific
-piece of data we want to display!
+piece of data we want to display! We can interpolate the `id` or our specific
+user into a `fetch` request URL and `fetch` that user's specific information
+from our backend:
 
 ```JavaScript
-const user = users.find(user => user.id === parseInt(params.id));
+const [user, setUser] = useState({});
+const params = useParams();
+const userId = params.id;
+
+useEffect(() =>{
+  fetch(`http://localhost:4000/users/${userId}`)
+  .then(r => r.json())
+  .then(data => setUser(data))
+  .catch(error => console.error(error))
+}, [userId]);
+
 ```
 
-We need to use `parseInt` here because all data passed via URL params will be
-formatted as strings.
+We'll also want to add some conditional rendering to make sure our app doesn't
+error out while it's waiting for our user to be fetched:
 
-> Note: In this example, we're finding the user we need in a list we imported
-> from the `data.js` file, but normally that will not be the case. If your data
-is contained in a `db.json` file or database instead, you'll likely run a
-`fetch` request to grab the specific piece of data you want from your database.
+```jsx
+if(!user.name){
+  return <h1>Loading...</h1>;
+};
+```
 
-Now that we have a way to access the user we want, let's update our
-`UserProfile` component to display information about that user!
+Here's what our full UserProfile component looks like now:
```

Comment:
Suggest adding just before this line:

Finally, let's update the jsx to display the user's name:

```jsx
<h1>{user.name}</h1>
```